### PR TITLE
fix: only expand curly env vars

### LINKED ISSUE #3669: Unable to connect when connection string contains $
**Present only in Prisma 2.7.0**

🚨 Affects all users, not only the ones using nested env variables.
```env
DATABASE_URL="postgres://user:password@server.host:5432/database?ssl=1&schema=schema$1234"
```

## Bug description

Edge case missing from tests:
If the connection string specifies a schema containing the symbol `$`, only the first component is actually parsed.

Probably not the best idea to use $ in a schema name, Prisma 1 naming is to blame !

## How to reproduce

Use the following `.env`

```env
DOTENV_PRISMA_EXPAND_DATABASE_URL="postgres://user:password@server.host:5432/database"
DOTENV_PRISMA_EXPAND_DATABASE_URL_WITH_SCHEMA="${DOTENV_PRISMA_EXPAND_DATABASE_URL}?ssl=1&schema=schema$1234"
```

Error displayed when performing ANY query (`schema` detected instead of `schema$1234`):
> Error: The table `schema.TableNameHere` does not exist in the current database.

## Expected behavior

Prisma should connect to the specified server, db and schema.

## Prisma information

N/A

## Environment & setup

- OS: 
- Database: 
- Node.js version: 
- Prisma version: 2.7.0

## Notes

The culprit is, most likely, `dotenv-expand`'s parsing (introduced in https://github.com/prisma/prisma/pull/3330).

### PR DESCRIPTION
**Pull request by timsuchanek**
Fixes #3669

### CODE REVIEW

File: src/packages/engine-core/src/NodeEngine.ts [hunk 1]
```diff
@@ -570,6 +570,7 @@ ${chalk.dim("In case we're mistaken, please report this to us 🙏.")}`)
 
         this.port = await this.getFreePort()
         const env = await this.getEngineEnvVars()
+        debug('env', env)
```
Comment (Jolg42) [inline src/packages/engine-core/src/NodeEngine.ts:573]: Do you want to keep this? It's outputting potential secrets to the logs when activated.

Comment (timsuchanek) [reply src/packages/engine-core/src/NodeEngine.ts:573]: Good point, will remove it again

**Review by Jolg42 — APPROVED** [review]
👍
